### PR TITLE
Fix regression where unlabeled files are not being identified

### LINKED ISSUE #822: Check for unlabeled files no longer works
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When I run validate it no longer flags unlabled-files, even if the --allow-unlabled-files option is not called.

### 🕵️ Expected behavior

I expect that when I run validate that it will warn me when there are files that do not have labels, as prior versions did.

### 📜 To Reproduce

1. Grab a bundle or collection and ensure there is an unlabeled file in the directory.
2. Run validate.  Example:  validate -R pds4.bundle -t [point to bundle file OR bundle directory]
3.
...

### 🖥 Environment Info

- Version of this software tested: 3.3.3 and 3.4.1 (note it works in version 3.3.0 and prior)
- Operating Systems: MacOSX 12.6.8 and Red Hat Enterprise Linux 9.3
...

### 📚 Version of Software Used

gov.nasa.pds:validate
Version 3.4.1
Release Date: 2023-12-14 00:22:59

Copyright 2019, by the California Institute of Technology ("Caltech").
All rights reserved.

gov.nasa.pds:validate
Version 3.3.3
Release Date: 2023-11-20 23:43:57

Copyright 2019, by the California Institute of Technology ("Caltech").
All rights reserved.

### 🩺 Test Data / Additional context

_No response_

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Put some code back in that got lost during the saxon upgrade #734. Probably removed in an attempt to make something work but it was not the cause and then left it out since none of the regression tests failed.

## ⚙️ Test Data and/or Report
See automated tests of this PR as they should pass

## ♻️ Related Issues
Closes #822

### CODE REVIEW

**Comment by al-niessner** [comment]
@jordanpadams 

Which 7 files do we expect to be unlabeled?

**Comment by jordanpadams** [comment]
@al-niessner not positive if these are referenced in a label or not, but I don't have time to look through of the labels to verify. Here is my guess (it may only be 6...):
* src/test/resources/github6/invalid//data_spectra/000DANGLING_FILE_WITH_EXTENSION.txt
* src/test/resources/github6/invalid//data_spectra/000DANGLING_FILE_WITHOUT_EXTENSION
* src/test/resources/github6/invalid//core
* src/test/resources/github6/invalid//file_aux_should_be_ok.tab
* src/test/resources/github6/invalid//core.vmac
* src/test/resources/github6/invalid//file_with_bad_char_#_.txt

**Comment by al-niessner** [comment]
> @al-niessner not positive if these are referenced in a label or not, but I don't have time to look through of the labels to verify. Here is my guess (it may only be 6...):
> 
> * src/test/resources/github6/invalid//data_spectra/000DANGLING_FILE_WITH_EXTENSION.txt
> * src/test/resources/github6/invalid//data_spectra/000DANGLING_FILE_WITHOUT_EXTENSION
> * src/test/resources/github6/invalid//core
> * src/test/resources/github6/invalid//file_aux_should_be_ok.tab
> * src/test/resources/github6/invalid//core.vmac
> * src/test/resources/github6/invalid//file_with_bad_char_#_.txt

@jordanpadams I am going to move away from this test to #597 and PR #615 because it is much cleaner and potentially related. Long post coming in near future saying no, but still cleaner for testing.

**Comment by al-niessner** [comment]
The code responsible for throwing this error is in one place and one place only:

https://github.com/NASA-PDS/validate/blob/918da20f5641443a8ceb146d97f327db2b1e0447/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/FindUnreferencedFiles.java#L86-L110

Blames says that the only change was a year ago in #615. It is possible that we now have false negatives but need to be very careful. One, it is looking only for targets (labels) that it found but are not used by aggregates. It will not find any random extra file because of the loop:

https://github.com/NASA-PDS/validate/blob/918da20f5641443a8ceb146d97f327db2b1e0447/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/FindUnreferencedFiles.java#L97

The loop function in question is here:

https://github.com/NASA-PDS/validate/blame/918da20f5641443a8ceb146d97f327db2b1e0447/src/main/java/gov/nasa/pds/tools/validate/InMemoryRegistrar.java#L165

and has not changed significantly in 5 years (did code reformat 2 years ago). My next guess is that some code block that used to use that problem type was axed. Therefore I rolled back to v3.3.0 which the user reported as working. Still just the one location (FindUnreferencedFiles). Interestingly, v3.3.0 contains #615.

It means that either it never detected any random file or what we stuffed in as a target could have been regular files if not referenced by a label. Preparing an approach for this testing but it will not be simple.

File: src/main/java/gov/nasa/pds/tools/validate/rule/RegisterTargets.java [hunk 1]
```diff
@@ -48,6 +49,9 @@ public void registerTargets() {
       try {
         Crawler crawler = getContext().getCrawler();
         WildcardOSFilter fileFilter = getContext().getFileFilters();
+        if (!"PDS4 Directory".equalsIgnoreCase(getContext().getRule().getCaption())) {
```
Comment (al-niessner) [inline src/main/java/gov/nasa/pds/tools/validate/rule/RegisterTargets.java:52]: This got removed in #734. Not sure why and did not leave myself a good comment. Without it, the filter is looking for just XML files.

**Review by nutjob4life — APPROVED** [review]

**Review by jordanpadams — APPROVED** [review]

**Pull request merged**